### PR TITLE
Adding Register and Login with Email.

### PR DESCRIPTION
**Pull request by perkinsjr**
The following is to resolve issue #179 

A quick explanation of the changes:

Added a new vue view called Register which handles the creation of a new user. 

JS handles the auth and then after an update of the Photo and Display name is added, if its null it will do nothing but it shows us how to add a new photo / display name.

On the login page, two inputs were added for email and password and a new button that handles auth via email and password. 

This should be a good start, let me know what you think and what we could do / need to do.

### CODE REVIEW

**Comment by kefranabg** [comment]
Hi @perkinsjr , Thank you for this PR ! We'll take time soon to review it !

**Comment by perkinsjr** [comment]
Added check for windows to this pull request. This fixes #191 and now will work on all platforms.

File: internals/helpers/git-helper.js [hunk 1]
```diff
@@ -45,7 +46,11 @@ async function checkIfRepositoryIsCleanable() {
  * @returns {Promise<any>}
  */
 async function removeGitRepository() {
-  return exec('rm -Rf .git')
+  if (os.platform() === 'win32') {
```
Comment (kefranabg) [inline internals/helpers/git-helper.js]: Could you do this change in a separate PR ? It's would be easier for me :) 
Why don't just do a rimraf of the `.git` folder ? It would avoid doing an if / else statement

Comment (perkinsjr) [reply internals/helpers/git-helper.js]: I can change it to a seperate PR for sure.

I can also change it to the rimraf it felt a bit over kill for a single command. Entirely your decision. Let me know

Comment (kefranabg) [reply internals/helpers/git-helper.js]: The advantage of using `rimraf` is that we can be sure that it will work on any platform :)

Comment (perkinsjr) [reply internals/helpers/git-helper.js]: Seems like that is the plan. I will get it back today.

Comment (kefranabg) [reply internals/helpers/git-helper.js]: Thank you so much !

**Comment by MaeseppTarvo** [comment]
@perkinsjr I think it is also important to "merge" multiple providers. What I mean is that if user is already signed up using Email/Password and now authenticates with Google. Then the user can't login anymore with Email and Password.

Useful links: https://firebase.google.com/docs/auth/web/email-link-auth , https://firebase.google.com/docs/auth/web/account-linking

**Comment by perkinsjr** [comment]
> I think it is also important to "merge" multiple providers. What I mean is that if user is already signed up using Email/Password and now authenticates with Google, then the user can't login anymore with Email and Password.

While I don't disagree with this. That has to be an architectural decision from @kefranabg @tbetous  as they will have to decide how to link the users in the storage solution they are offering. Which is something we would have to have a back and forth about to make sure that it is the most flexible solution as we don't want to pigeon hole ourselves into a solution that works but becomes difficult to flex for new developers or developers who haven't used either bento-starter or authentication from firebase .

**Comment by MaeseppTarvo** [comment]
@perkinsjr I totally agree on that.

**Comment by kefranabg** [comment]
Discussed in #179
